### ISSUE TITLE
hamcrest matchers should be in separate module

### ISSUE THREAD

**Issue by neonailol** [yegor256/cactoos#751]
It's nice that cactoos have its own set of hamcrest matchers, but it pollutes classpath, and ideologically should be in separate module eg `cactoos-matchers`

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by ashtonhogan** [comment]
alternatively, you could use [Tacinga](https://tacinga.com) which already does this

**Comment by yegor256** [comment]
@llorllale I think it's a reasonable request, to be honest...

**Comment by llorllale** [comment]
@yegor256 sounds reasonable. I believe the new project, `cactoos-matchers`, should be brought under the same ZC umbrella [C63314D6Z](https://www.0crat.com/p/C63314D6Z).

My doubts revolve around the new project's CD: the ARC is [responsible](http://www.yegor256.com/2015/08/04/nine-steps-start-software-project.html) for this, but this one would seem like it needs your keys.

Please confirm.

**Comment by yegor256** [comment]
@llorllale two options 1) I create it in my GitHub account and deploy to Maven Central into `org.cactoos` group; 2) you create it in your GitHub account and deploy it to Maven yourself, if you a group there. In either case we add it to ZC project (you remain the ARC). Which option you prefer?

**Comment by llorllale** [comment]
@yegor256 I'll take care of it with option 2

**Comment by llorllale** [comment]
@0crat in

**Comment by llorllale** [comment]
@0crat assign @llorllale

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/yegor256/cactoos/issues/751#issuecomment-383184147))

@llorllale Job [#751](https://github.com/yegor256/cactoos/issues/751) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @neonailol[/z](https://www.0crat.com/u/neonailol)

**Comment by 0crat** [comment]
> @0crat assign @llorllale ([here](https://github.com/yegor256/cactoos/issues/751#issuecomment-383184179))

@llorllale The job [#751](https://github.com/yegor256/cactoos/issues/751) assigned to @llorllale[/z](https://www.0crat.com/u/llorllale), here is [why](https://www.0crat.com/footprint/C63314D6Z/1101840401); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html)

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by llorllale** [comment]
@0crat waiting

**Comment by 0crat** [comment]
> @0crat waiting taking longer than anticipated ([here](https://github.com/yegor256/cactoos/issues/751#issuecomment-385270265))

@llorllale The impediment for [#751](https://github.com/yegor256/cactoos/issues/751) was registered successfully by @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by llorllale** [comment]
@yegor256 do you wish to include it in ZC project C63314D6Z?

**Comment by yegor256** [comment]
@llorllale you can do it yourself in Slack

**Comment by llorllale** [comment]
@yegor256 that's correct, but unfortunately it says [here](https://www.yegor256.com/2014/07/31/travis-and-rultor.html#scenario-4-deploy-to-cloudbees) that rultor pulls secrets/keys from GH repos. Unfortunately, I don't have private repos and can't currently commit to spending those $7/mo due to special circumstances.

I believe having this non-rultor project next to the others would be awkward. I can assign this task to you so you can set it up with rultor. WDYT?

**Comment by llorllale** [comment]
@yegor256 forget it. I've thought about it a bit more and I've decided to get the subscription. Thanks for your patience.

**Comment by llorllale** [comment]
@0crat waiting on yegor256/rultor#1256

**Comment by 0crat** [comment]
> @0crat waiting on yegor256/rultor#1256 ([here](https://github.com/yegor256/cactoos/issues/751#issuecomment-387451300))

@llorllale Job [#751](https://github.com/yegor256/cactoos/issues/751) is already on hold

**Comment by llorllale** [comment]
@neonailol fixed in 6a2e21005ec29ba3d271f9b9b8b67f8352f8e23e

**Comment by llorllale** [comment]
@rultor release, tag is `0.33`

**Comment by llorllale** [comment]
@yegor256 @neonailol the new project is `llorllale/cactoos-matchers`.

**Comment by llorllale** [comment]
@yegor256 please checkout the license on the new project and feel free to open an issue in that project if applicable

**Comment by yegor256** [comment]
@llorllale looks good to me
